contracts/FundDistribution.sol

```diff
@@ -0,0 +1,257 @@
+    address to,
+    uint256 amount
+  ) external {
+    depositToken(token, amount);
+    setTokenApprove(to, token, amount);
+  }
+
+  //set the amount claimable to an address
+  function setEthApprove(address to, uint256 amount)
+    public
+    onlyDistributors
+    onlyNonZeroAddress(to)
+  {
+    ethAvailable[to] = amount;
+    emit EthApproveIsSet(to, amount);
+  }
+
+  function setEthApproveMultiple(address[] calldata tos, uint256[] calldata amounts) external {
+    require(tos.length == amounts.length, "Invalid array lengths");
+    for (uint256 i = 0; i < tos.length; i++) {
+      setEthApprove(tos[i], amounts[i]);
+    }
+  }
+
+  //set the token amount claimable to an address
+  function setTokenApprove(
+    address to,
+    address token,
+    uint256 amount
+  ) public onlyDistributors onlyNonZeroAddress(to) {
+    require(curTokens[token], "Token is not added");
+    tokenAvailable[to][token] = amount;
+    emit TokenApproveIsSet(to, token, amount);
+  }
+
+  function setTokenApproveMultiple(
+    address[] calldata tos,
+    address tokensApprove,
+    uint256[] calldata amounts
+  ) external {
+    require(tos.length == amounts.length, "Invalid array lengths");
+    for (uint256 i = 0; i < tos.length; i++) {
+      setTokenApprove(tos[i], tokensApprove, amounts[i]);
+    }
+  }
+
+  //the sender claim his ether, not revert if insufficient ether
+  function claimEth() public {
+    sendEthTo(msg.sender);
+  }
+
+  function claimEthWithRevertIfInsufficientFunds() public payable {
+    sendEthToWithRevertIfInsufficientFunds(msg.sender);
+  }
+
+  //the sender claim his token, not revert if insufficient funds
+  function claimToken(address token) external {
+    sendTokenTo(msg.sender, token);
+  }
+
+  function claimTokenWithRevertIfInsufficientFunds(address token) public payable {
+    sendTokenToWithRevertIfInsufficientFunds(msg.sender, token);
+  }
+
+  //the sender claim all his funds, not revert if insufficient funds
+  function claimAllFunds() external {
+    sendAllFundsTo(msg.sender);
+  }
+
+  function claimAllFundsWithRevertIfInsufficientFunds() public payable {
+    sendAllFundsToWithRevertIfInsufficientFunds(msg.sender);
+  }
+
+  //claim eth on behalf of an address, not revert if insufficient funds
+  function sendEthTo(address to) public onlyNonZeroAddress(to) {
+    uint256 amount = _min(ethAvailable[to], address(this).balance);
+    _transferEth(to, amount);
+  }
+
+  function sendEthToWithRevertIfInsufficientFunds(address to) public {
+    _transferEth(to, ethAvailable[to]);
+  }
+
+  //claim token on behalf of an address, not revert if insufficient funds
+  function sendTokenTo(address to, address token)
+    public
+    onlyNonZeroAddress(to)
+    onlyNonZeroAddress(token)
+  {
+    uint256 amount = _min(tokenAvailable[to][token], IERC20(token).balanceOf(address(this)));
+    _transferToken(to, token, amount);
+  }
+
+  function sendTokenToWithRevertIfInsufficientFunds(address to, address token) public {
```

Comment:
Can be done with one require

Reply:
yes I have one require below, I just offer generous option to withdraw for users

Reply:
yes what i mean is return boll and use require in exactly place then can prevent write too much function